src/resources/2021/DaylightShavings.ts

```diff
@@ -0,0 +1,66 @@
+import { myClass, toEffect, toInt } from "kolmafia";
+import { have as haveItem } from "../../lib";
+import { get } from "../../property";
+import { $effects, $item } from "../../template-string";
+
+export const helmet = $item`Daylight Shavings Helmet`;
+
+/**
+ * Returns whether the player owns an unpackaged Daylight Shavings Helmet, and it's available in either the inventory or other zones as determined by autoSatisfy settings.
+ * @returns whether we have the Daylight Shavings Helmet.
+ */
+export function have(): boolean {
+  return haveItem(helmet);
+}
+
+export const buffs = $effects`Spectacle Moustache, Toiletbrush Moustache, Barbell Moustache, Grizzly Beard, Surrealist's Moustache, Musician's Musician's Moustache, Gull-Wing Moustache, Space Warlord's Beard, Pointy Wizard Beard, Cowboy Stache, Friendly Chops`;
+
+export function hasBuff(): boolean {
+  return buffs.some((buff) => haveItem(buff));
+}
+/**
+ * Calculates and returns the cycle of buffs that the hat should cycle through.
+ * @param playerclass The class to generate a cycle for
+ * @returns An ordered array consisting of the cycle for this class. The first element of the array will be the first buff a player should expect to get in a given ascension.
+ */
+export function buffCycle(playerclass = myClass()): Effect[] {
```

Comment:
my assumption is the answer to this is "i dunno, sue me" but do we know what happens to shaving helmet cycle when you drop path? that is the one place i could see this breaking, and while it doesn't seem like a major vector for bugs, it is probably worth at least doing minor digging to see if this is a possible breakpoint for libram, or putting a catch here to output a null cycle the day you have dropped path or ascended out of an avatar.

Reply:
If someone wants to spade this, I will happily update the code. I've made sure it no longer explodes if you somehow submit astral spirit, but I don't think we need to independently check this out before publishing.